azurelinuxagent/ga/exthandlers.py

```diff
@@ -150,8 +150,13 @@ def parse_ext_substatus(substatus):
 
 
 def parse_ext_status(ext_status, data):
-    if data is None or len(data) == 0: # pylint: disable=len-as-condition
+    if data is None:
         return
+    if not isinstance(data, list):
+        raise ExtensionStatusError(msg="The extension status must be an array: {0}".format(data), code=ExtensionStatusError.StatusFileMalformed)
+    if not data:
```

Comment:
NIT: Redundant check (line 153 and 157)

Reply:
it not redundant: the first check is for None, the second for length of the list (somehow PEP 8 doesn't like an explicit comparison len(foo) == 0 and asks for not foo (that is the reason for the pylint warning in the previous code)

btw - our code and unit tests allow for an empty status, which i get the feeling is not the right behavior. I decided to keep that behavior for now, I may come back to it late.

Reply:
>btw - our code and unit tests allow for an empty status, which i get the feeling is not the right behavior. I decided to keep that behavior for now, I may come back to it later.

Ahh I see, no wonder you needed 2 extra checks for it. Good catch though, make sure to open a tfs bug for it to ensure we dont forget :)